x/mongo/driver/compression.go

```diff
@@ -25,6 +25,32 @@ type CompressionOpts struct {
 	UncompressedSize int32
 }
 
+func calcZstdWindowSize(n int, l zstd.EncoderLevel) int {
+	windowSize := zstd.MinWindowSize
+	if n > windowSize {
+		// Map the window size with compression levels as the zstd package does.
+		switch l {
+		case zstd.SpeedFastest:
+			windowSize = 4 << 20
+		case zstd.SpeedDefault:
+			windowSize = 8 << 20
+		case zstd.SpeedBetterCompression:
+			windowSize = 16 << 20
+		case zstd.SpeedBestCompression:
+			windowSize = 32 << 20
+		}
+		if windowSize > zstd.MaxWindowSize {
+			windowSize = zstd.MaxWindowSize
+		}
+		// Reduce the window size to the closest power of 2 that can hold the input size
+		// if the default window size is larger than the input size.
+		for windowSize/2 > n {
```

Comment:
So it looks like this reduction is our custom piece of window size calculation, and using the smallest window size possible results in less memory being allocated? Could you point me to the analogue of `calcZstdWindowSize` in the `zstd` library?

Reply:
Sorry, I forgot to leave a reference.
The original code is [here](https://github.com/klauspost/compress/blob/master/zstd/encoder_options.go#L215).
And the instruction on the window size setting can be found [here](https://pkg.go.dev/github.com/klauspost/compress/zstd#WithWindowSize).